modules/core/src/effects/mask/mask-effect.js

```diff
@@ -149,6 +160,8 @@ export default class MaskEffect extends Effect {
       }
       channelInfo.layers.push(layer);
       channelInfo.layerBounds.push(layer.getBounds());
+      channelInfo.coordinateOrigin = layer.props.coordinateOrigin;
```

Comment:
Set this only once when the layer is created (the root layer's props)?

Technically a parent layer can have a different coordinate system from its children (e.g. MVT). To support that scenario we'll need to finish implementing the `getBounds` method for individual composite layers, instead of merging all layer bounds in this effect.

Reply:
I'm not sure if I correctly understood you here. Did you mean this: https://github.com/visgl/deck.gl/pull/6591/commits/1686fcd33ef913af07280aa77992f5c725038347

Reply:
I meant move this into the block `channelInfo = {...}`. You are re-assigning these two fields again and again for each descendant of any composite layer.